allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -39,9 +46,90 @@ def __init__(self, cache_dir, project_name):
+        # dict mapping paths to downloaded files to their file_hashes;
+        # this will be used when determining if a downloaded file
+        # can instead be a symlink
+        c_path = pathlib.Path(self._cache_dir)
+        self._downloaded_data_path = c_path / '_downloaded_data.json'
+
+        if not self._downloaded_data_path.exists():
+            self._construct_file_version_lookup()
+
+    def _construct_file_version_lookup(self) -> None:
+        """
+        Construct the dict that maps between file_hash and
+        absolute local path. Save it to self._downloaded_data_path
+        """
+        lookup = {}
+        for mfest_name in self.list_all_downloaded_manifests():
+
+            # Call the private method so that we don't accidentally
+            # raise the "a more up to date version of the manifest
+            # exists" warning. That is not the point here.
+            self._manifest = self._load_manifest(mfest_name)
+
+            for file_id in self._manifest.file_id_values:
+                attr = self.data_path(file_id)
+                if attr['exists']:
+                    local_path = str(attr['local_path'].resolve())
+                    hsh = attr['file_attributes'].file_hash
+                    lookup[local_path] = hsh
+
+            for metadata_name in self._manifest.metadata_file_names:
+                attr = self.metadata_path(metadata_name)
+                if attr['exists']:
+                    local_path = str(attr['local_path'].resolve())
+                    hsh = attr['file_attributes'].file_hash
+                    lookup[local_path] = hsh
+
+        with open(self._downloaded_data_path, 'w') as out_file:
+            out_file.write(json.dumps(lookup, indent=2, sort_keys=True))
+
+        self._manifest = None
+
+    def _warn_of_outdated_manifest(self, manifest_name: str) -> None:
+        """
+        Warn that manifest_name is not the latest manifest available
+        """
+        if self._manifest_last_warned_on is not None:
+            if self.latest_manifest_file == self._manifest_last_warned_on:
+                return None
+
+        self._manifest_last_warned_on = self.latest_manifest_file
+
+        msg = '\n\n'
+        msg += 'The manifest file you are loading is not the '
+        msg += 'most up to date manifest file available for '
+        msg += 'this dataset. The most up to data manifest file '
+        msg += 'available for this dataset is \n\n'
+        msg += f'{self.latest_manifest_file}\n\n'
+        msg += 'To see the differences between these manifests,'
+        msg += 'run\n\n'
+        msg += f"self.compare_manifests('{manifest_name}', "
+        msg += f"'{self.latest_manifest_file}')\n\n"
+        msg += "To see all of the manifest files currently downloaded "
+        msg += "onto your local system, run\n\n"
+        msg += "self.list_all_downloaded_manifests()\n\n"
+        warnings.warn(msg, OutdatedManifestWarning)
```

Comment:
I think this warning should also instruct "here's what to do to update to the latest"

Reply:
Done